src/FixedFrequencyAgent.cpp

```diff
@@ -0,0 +1,259 @@
+            if (!std::isnan(uncore_min_freq_request)) {
+                m_platform_io.write_control("CPU_UNCORE_FREQUENCY_MIN_CONTROL",GEOPM_DOMAIN_BOARD,0,uncore_min_freq_request);
+            }
+
+            if (!std::isnan(uncore_max_freq_request)) {
+                m_platform_io.write_control("CPU_UNCORE_FREQUENCY_MAX_CONTROL",GEOPM_DOMAIN_BOARD,0,uncore_max_freq_request);
+            }
+
+            m_is_adjust_initialized = true;
+        }
+
+    }
+
+    //If new values have been adjusted, write
+    bool FixedFrequencyAgent::do_write_batch(void) const
+    {
+        return false;
+    }
+
+    // Read signals from the platform and calculate samples to be sent up
+    void FixedFrequencyAgent::sample_platform(std::vector<double> &out_sample)
+    {
+
+    }
+
+    // Wait for the remaining cycle time to keep Controller loop cadence at 1 second
+    void FixedFrequencyAgent::wait(void)
+    {
+        geopm_time_s current_time;
+        do {
+            geopm_time(&current_time);
+        }
+        while(geopm_time_diff(&m_last_wait, &current_time) < M_WAIT_SEC);
+        geopm_time(&m_last_wait);
+    }
+
+    // Adds the wait time to the top of the report
+    std::vector<std::pair<std::string, std::string> > FixedFrequencyAgent::report_header(void) const
+    {
+        return {};
+    }
+
+    // Adds to the per-node section of the report
+    std::vector<std::pair<std::string, std::string> > FixedFrequencyAgent::report_host(void) const
+    {
+        return {};
+    }
+
+    // This Agent does not add any per-region details
+    std::map<uint64_t, std::vector<std::pair<std::string, std::string> > > FixedFrequencyAgent::report_region(void) const
+    {
+        return {};
+    }
+
+    std::vector<std::string> FixedFrequencyAgent::trace_names(void) const
+    {
+        std::vector<std::string> names;
+        return names;
+    }
+
+    // Updates the trace with values for samples and signals from this Agent
+    void FixedFrequencyAgent::trace_values(std::vector<double> &values)
+    {
+
+    }
+
+    std::vector<std::function<std::string(double)> > FixedFrequencyAgent::trace_formats(void) const
+    {
+        std::vector<std::function<std::string(double)>> trace_formats;
+        return trace_formats;
+    }
+
+    // Name used for registration with the Agent factory
+    std::string FixedFrequencyAgent::plugin_name(void)
+    {
+        return "fixed_frequency";
+    }
+
+    // Used by the factory to create objects of this type
+    std::unique_ptr<Agent> FixedFrequencyAgent::make_plugin(void)
+    {
+        return geopm::make_unique<FixedFrequencyAgent>();
+    }
+
+    // Describes expected policies to be provided by the resource manager or user
+    std::vector<std::string> FixedFrequencyAgent::policy_names(void)
+    {
+        return {"GPU_FREQUENCY",
+                "CORE_FREQUENCY",
+                "UNCORE_MIN_FREQUENCY",
+                "UNCORE_MAX_FREQUENCY",
+                "SAMPLE_PERIOD",
```

Comment:
New direction for post-2.0 is to not provide sample_period as part of the policy, but to do so via the geopmlaunch command.   We should remove SAMPLE_PERIOD from all post-2.0 agent.s

Reply:
Got it. Thanks for the note. So this is the whole deal with the sample period thing that was mentioned earlier when we reviewed the PR.